### PR TITLE
fix: [CDS-40944]: yamlToVisual duplicate identifier condition update + YamlBuilder flicker issue

### PR DESCRIPTION
**Pull request by PratyushGar**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 5 Issues 
* Code issue 5 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10519)

**Review by vkbansal-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/PipelineCanvas.tsx [hunk 1]
```diff
@@ -722,24 +721,18 @@ export function PipelineCanvas({
...
-      const duplicateStepIdentifiersList = [] as string[]
-      Object.values(stepsIdMap).forEach(stepIdMap => {
-        const duplicateSteps = hasDuplicateIdentifier(stepIdMap.steps)
-        const duplicateRollbackSteps = hasDuplicateIdentifier(stepIdMap.rollbackSteps)
-        duplicateSteps.value && duplicateStepIdentifiersList.push(...duplicateSteps.duplicateIds)
-        duplicateRollbackSteps.value && duplicateStepIdentifiersList.push(...duplicateRollbackSteps.duplicateIds)
-      })
+      const duplicateStepIdentifiersList = getDuplicateStepIdentifierList(stagesData)
       duplicateStepIdentifiersList.length &&
         showError(
```
Comment (yc147) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/PipelineCanvas.tsx:726]: Not the right place to show the error.

Comment (PratyushGar) [reply src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/PipelineCanvas.tsx:726]: Added here as duplicate Identifiers was also being added in error message.

**Review by yc147 — CHANGES_REQUESTED** [review]
Not the right place to show the error. This function is supposed to run a boolean. Moved the logic to show error out side this function.

File: src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/PipelineCanvasUtils.ts [hunk 2]
```diff
@@ -163,3 +164,15 @@ export const getStageIdDetailsMapping = (stagesData: StageElementWrapperConfig[]
   })
   return stageStepsMapping
 }
+
+export const getDuplicateStepIdentifierList = (stagesData: StageElementWrapperConfig[]): string[] => {
+  const duplicateStepIdentifiersList = [] as string[]
+  const stepsIdMap = getStageIdDetailsMapping(stagesData)
+  Object.values(stepsIdMap).forEach(stepIdMap => {
+    const duplicateSteps = hasDuplicateIdentifier(stepIdMap.steps)
```
Comment (yc147) [inline src/modules/70-pipeline/components/PipelineStudio/PipelineCanvas/PipelineCanvasUtils.ts:172]: `hasDuplicateIdentifier` has need to be removed. Name suggests it should return a boolean but it actually returns object.

**Review by yc147 — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**